pytups/tuplist.py

```diff
@@ -100,14 +100,33 @@ def to_dict(
         """
         from . import superdict as sd
 
-        if result_col is None:
+        if len(self) == 0:
+            return sd.SuperDict()
+        # Handle case of list of dict
+        if isinstance(self[0], dict):
+            if indices is None:
+                raise ValueError(
+                    "For a list of dict, to_dict require indices to be specified"
+                )
+            if result_col is None:
+                if type(indices) is not list or len(indices) == 1:
```

Comment:
I do not  like checking for a list of size 1. If a list is passed, even it has size 1, then a tuple needs to be returned.

Reply:
also I do not like checking for `type() is not list`. Although I have seen this is what exists in the `to_dict` too. I have a function in tools.py that is called `is_really_iterable` that I think should be used in both places.

Reply:
tuple of size 1 do no really make sense.
If you pass [1] as an argument, you do not expect to have (1,) as a key of your dict. You expect 1.

I will make the change with is_really_iterable

Reply:
I also wonder if for dict, we should consider that the default value of 0 for result_col is the same as None (because 90% it will result in an error hard to understand.
Maybe put this before the if result_col is None:
```
if result_col == 0 and 0 not in self:
    result_col=None
```